packages/ui/src/components/widgets/SwapWidgetRenderer.tsx

```diff
@@ -766,40 +771,36 @@ const SwapWidgetRenderer: FC<SwapWidgetRendererProps> = ({
           : error
       )
 
-      onAnalyticEvent?.(EventNames.SWAP_ERROR, {
-        error_message: errorMessage,
-        wallet_connector: connector?.name,
-        quote_id: extractQuoteId(steps ?? (quote?.steps as Execute['steps'])),
-        amount_in: parseFloat(`${debouncedInputAmountValue}`),
-        currency_in: fromToken?.symbol,
-        chain_id_in: fromToken?.chainId,
-        amount_out: parseFloat(`${debouncedOutputAmountValue}`),
-        currency_out: toToken?.symbol,
-        chain_id_out: toToken?.chainId,
-        is_canonical: useExternalLiquidity,
-        txHashes: (steps ?? (quote?.steps as Execute['steps']))
-          ?.map((step) => {
-            let txHashes: { chainId: number; txHash: string }[] = []
-            step.items?.forEach((item) => {
-              if (item.txHashes) {
-                txHashes = txHashes.concat([
-                  ...(item.txHashes ?? []),
-                  ...(item.internalTxHashes ?? [])
-                ])
-              }
-            })
-            return txHashes
-          })
-          .flat()
-      })
+      const { step, stepItem } = getCurrentStep(currentSteps)
+      const swapEventData = getSwapEventData(
+        quote?.details,
+        currentSteps ?? null,
+        connector
+      )
+      const isApproval = step?.id === 'approve'
```

Comment:
I think we need to check if the step.id is also either an `approval` or `authorize`

Reply:
Going to punt on authorize but I will add approval